sdk/tools/azure-openrewrite-compiler-maven-plugin/src/main/java/com/azure/openrewrite/goldenimages/GoldenImageCompilerMojo.java

```diff
@@ -0,0 +1,252 @@
+
+    @Parameter(defaultValue = "${session}", readonly = true, required = true)
+    private MavenSession session;
+
+    @Component
+    private ProjectBuilder projectBuilder;
+
+    public void execute() throws MojoExecutionException {
+        String testResourcesPath = baseDir.getAbsolutePath() + "/src/test/resources/migrationExamples";
+        File testResourcesDir = new File(testResourcesPath);
+
+        if (!testResourcesDir.exists()) {
+            throw new MojoExecutionException("Test resources directory does not exist: " + testResourcesPath);
+        }
+
+        List<File> testFolders = findTestFolders(testResourcesDir);
+        String[] profiles = targetProfiles.split(",");
+
+        for (File folder : testFolders) {
+            for (String profile : profiles) {
+                File profileDir = new File(folder, profile);
+
+                if (!profileDir.exists()) {
+                    getLog().warn("Skipping " + folder.getAbsolutePath() + " as profile directory " + profile + " does not exist.");
+                    continue;
+                }
+
+                File previousProfileDir = getPreviousProfileDir(folder, profile);
+                if (previousProfileDir != null && previousProfileDir.exists()) {
+                    if (hasChanges(profileDir, previousProfileDir)) {
+                        compileJavaFilesWithProfile(profileDir, profile);
+                    } else {
+                        getLog().info("No changes detected between " + profileDir.getAbsolutePath() + " and " + previousProfileDir.getAbsolutePath());
+                    }
+                } else {
+                    compileJavaFilesWithProfile(profileDir, profile);
+                }
+            }
+        }
+    }
+
+    private List<File> findTestFolders(File baseDir) throws MojoExecutionException {
+        try (Stream<Path> paths = Files.walk(baseDir.toPath())) {
+            return paths
+                .filter(Files::isDirectory)
+                .filter(path -> Arrays.stream(targetProfiles.split(","))
+                    .allMatch(profile -> path.resolve(profile).toFile().exists()))
+                .map(Path::toFile)
+                .collect(Collectors.toList());
+        } catch (IOException e) {
+            throw new MojoExecutionException("Failed to find test folders", e);
+        }
+    }
+
+    private File getPreviousProfileDir(File folder, String profile) {
+        int profileVersion;
+        try {
+            profileVersion = Integer.parseInt(profile.substring(1));
+        } catch (NumberFormatException e) {
+            return null;
+        }
+        if (profileVersion > 1) {
+            return new File(folder, "v" + (profileVersion - 1));
+        }
+        return null;
+    }
+
+    private boolean hasEqualFileContents(File file1, File file2) throws IOException {
+
+
+        String before = Files.readAllLines(file1.toPath())
+            .stream()
+            .reduce("", (a, b) -> a + b + "\n");
```

Comment:
It might be easier to read if we use the collector instead of `reduce`.
```java
Files.readAllLines(..)
       .stream()
       .collect(Collectors.joining("\n"));
```